### DISCUSSION TITLE
关于arxiv论文网站翻译的几点建议，希望采纳

### DISCUSSION THREAD

**Discussion by zhufeizzz** [immersive-translate/immersive-translate#1777] [Ideas]
### 新功能 | New Feature

1. 现在翻译过慢，滚动一下经常卡死。建议已翻译的论文进行全部缓存，提升效率。
2. 目前经常翻译无意义的内容，包括表格、代码和论文引用信息，这些东西完全没必要翻译，很影响观感。
3. 建议翻译后能提供论文章节导航目录，方便跳转。
4. 建议提供左右对照的翻译视图，并且支持选中语句高亮。现在上下对照看起来不方便，参考一译网站。
5. 现在好像没看到按论文版本翻译的按钮，也可能是我没找到。

### 为什么您想要添加这些功能 | Why you want to add these features

NA

### 提议 | Proposal

翻译效果挺好的，建议增强arxiv网站的体验。

**Comment by theowenyoung** [comment]
> 现在翻译过慢，滚动一下经常卡死。建议已翻译的论文进行全部缓存，提升效率。

请问是在什么设备和浏览器上经常卡死？有没有示例的链接？ 卡死理论上和 缓存 没有关系，插件本地如果翻译过，目前是会缓存的。

> 目前经常翻译无意义的内容，包括表格、代码和论文引用信息，这些东西完全没必要翻译，很影响观感。

请问是否有示例链接？可以做进一步优化。


> 建议翻译后能提供论文章节导航目录，方便跳转。

这个功能可能更适合某一个独立的插件或者脚本来做，这样沉浸式翻译本身可以保持轻量一点。

> 建议提供左右对照的翻译视图，并且支持选中语句高亮。现在上下对照看起来不方便，参考一译网站。

一译确实实现的很好，目前我们的基础设施并不支持按照句子级别做翻译，并且也不支持左右对照翻译，这个可能会放在未来研究。

> 现在好像没看到按论文版本翻译的按钮，也可能是我没找到。

请问是否有多版本的示例链接？ 理论上只要 Arxiv 本身提供了多版本的 html，沉浸式翻译便支持翻译。

**Reply by LeiShang2004** [reply]
您好，针对问题一：我的设备是联想拯救者Y9000P2022，在edge130.0.2849.68（有较多的扩展插件）和谷歌浏览器130.0.6723.117（基本没有扩展插件）两个浏览器上，arxiv论文网站翻译加载时均十分卡顿，浏览器卡死甚至操作系统一并卡死。

**Comment by zhufeizzz** [comment]
**问题1：现在翻译过慢，滚动一下经常卡死。建议已翻译的论文进行全部缓存，提升效率。**
现在无法重现了，可能是我电脑的问题，请忽略。

**问题2：目前经常翻译无意义的内容，包括表格、代码和论文引用信息，这些东西完全没必要翻译，很影响观感。**
期望：不要翻译代码、表格、论文引用。现在只能靠“翻译段落所需的最少字符数”配置来规避，但是效果一般。
![image](https://github.com/immersive-translate/immersive-translate/assets/4591255/a58f0536-0ef0-42c1-a909-dc4df9a52d41)
![image](https://github.com/immersive-translate/immersive-translate/assets/4591255/cdfaad8c-3fc3-4ad4-bd31-ce56aa6adc94)


**问题5：现在好像没看到按论文版本翻译的按钮，也可能是我没找到。**
试了下，在arXiv概要视图切换版本后是可以针对指定版本翻译的，是我搞错了，请忽略。

**问题6：再额外补充一个问题。**
现在插件悬浮球没法很好的识别中英文，导致每个网页都会显示悬浮球。实际我只想要arXiv网站显示即可，但现在的配置好像满足不了这种要求。

@theowenyoung

**Reply by theowenyoung** [reply]
可以提供一下 2 的链接吗？我们测试一下。 用最小字数也是一个不错的规避办法，我平时会设置为15.

> 现在插件悬浮球没法很好的识别中英文，导致每个网页都会显示悬浮球。实际我只想要arXiv网站显示即可，但现在的配置好像满足不了这种要求。

如果不想看到悬浮球，建议直接隐藏即可。使用快捷键或者 面板，右键作为触发方式。

**Reply by zhufeizzz** [reply]
https://ar5iv.labs.arxiv.org/html/2306.05301?_immersive_translate_auto_translate=1
